### PR TITLE
[FLINK-12720][python] Add the Python Table API Sphinx docs

### PR DESCRIPTION
**Pull request by HuangXingBo**
## What is the purpose of the change

As the Python Table API is added, we should add the Python Table API Sphinx docs. This includes the following work:
Add scripts to build the Sphinx docs

## Brief change log

  - *Support Sphinx to create Flink Python Docs*
  - *Correct doc code style of Python code*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

File: flink-python/docs/pyflink.table.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+pyflink.table package
+=====================
+
+Module contents
+---------------
+
+.. automodule:: pyflink.table
+    :members:
+    :undoc-members:
+    :show-inheritance:
```
Comment (dianfu) [inline flink-python/docs/pyflink.table.rst]: One empty line at the end of the file

File: flink-python/docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+# -- Project information -----------------------------------------------------
+
+# project = u'Flink Python Table API'
+project = u'PyFlink'
+copyright = u''
+author = u'Author'
+
+# The short X.Y version
+version = '1.0'
```
Comment (sunjincheng121) [inline flink-python/docs/conf.py]: The version should same as : https://github.com/apache/flink/blob/master/flink-python/pyflink/version.py

**Comment by sunjincheng121** [comment]
I check it again, I found that Java and Scala Doc are built by BuilderBot(https://ci.apache.org/projects/flink/), The BuilderBot need a config `flink.conf`, and this config pleased `svn` repo. So we can do the copy logic change in a new PR. :) @HuangXingBo

**Comment by HuangXingBo** [comment]
Thanks a lot for @dianfu @sunjincheng121.
1. I add logic about sphinx installed and check in lint-python.sh.
2. I add a generate-docs.sh responsible for call the script lint-python.sh to generate python api docs and copy the api docs to the specified target path.

File: flink-python/docs/generate-docs.sh [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            TARGET_PATH=$OPTARG
+            ;;
+        ?)
+            printf "ERROR: did not recognize option '%s', please try -h\\n" "$1"
+            exit 1
+            ;;
+    esac
+done
+
+LINTPYTHON_PATH="../dev/lint-python.sh"
```
Comment (dianfu) [inline flink-python/docs/generate-docs.sh]: The means that we must execute this script under directory doc.
What about changed as follows:
CURRENT_DIR="$(cd "$( dirname "$0" )" && pwd)"
LINTPYTHON_PATH="$CURRENT_DIR/../dev/lint-python.sh"

**Comment by sunjincheng121** [comment]
@dianfu Thanks for helping improve the PR! @HuangXingBo told me he is on the vacation cannot update the PR those days, he also agrees that if you want to help him, you can open a new PR with @HuangXingBo 's commit as a basic commit.  What do you think?

**Comment by dianfu** [comment]
@sunjincheng121 Thanks a lot for the suggestion. Makes sense to me and I have created PR #8774 based on @HuangXingBo 's great work.

**Comment by sunjincheng121** [comment]
Thanks for your effort @HuangXingBo! and Thanks for open the new PR @dianfu!
I close this PR and review the new one. :)

**Pull request closed without merging**